metashare/api/models.py

```diff
@@ -76,28 +79,46 @@ def full_org_type(self):
     @property
     def instance_url(self):
         try:
-            return self.social_account.extra_data["instance_url"]
+            return self.salesforce_account.extra_data["instance_url"]
         except (AttributeError, KeyError):
             return None
 
     @property
     def token(self):
-        account = self.social_account
+        account = self.salesforce_account
         if account and account.socialtoken_set.exists():
-            token = self.social_account.socialtoken_set.first()
-            return (fernet_decrypt(token.token), fernet_decrypt(token.token_secret))
+            token = self.salesforce_account.socialtoken_set.first()
+            try:
+                return (
+                    fernet_decrypt(token.token) if token.token else None,
+                    token.token_secret if token.token_secret else None,
+                )
+            except InvalidToken:
+                return (None, None)
         return (None, None)
 
     @property
-    def social_account(self):
-        return self.socialaccount_set.first()
+    def salesforce_account(self):
+        return self.socialaccount_set.filter(provider="salesforce-production").first()
 
     @property
     def valid_token_for(self):
         if all(self.token) and self.org_id:
             return self.org_id
         return None
 
+    @cached_property
+    def is_dev_hub_enabled(self):
+        token, _ = self.token
+        instance_url = self.salesforce_account.extra_data["instance_url"]
+        url = f"{instance_url}/services/data/v45.0/sobjects/ScratchOrgInfo"
+        resp = requests.get(url, headers={"Authorization": f"Bearer {token}"})
+        if resp.status_code == 200:
+            return True
+        if resp.status_code == 404:
+            return False
+        return None
```

Comment:
If we can determine the org is a sandbox or scratch org from the info that's stored in extra_data (see logic in metadeploy), we can short-circuit in that case and return False to avoid the API call.

Reply:
Looks like this property isn't used yet?

Reply:
It isn't used yet, no. I'd rather do it based on `extra_data` if we can, to avoid a nested HTTP call, but I don't know how.